music_this_week_app/backend/spotifyHandler.py

```diff
@@ -76,22 +109,39 @@ def filter_artist(self, artist_name):
                     print ("Exact match!")
                 else:
                     print ("Close enough...")
-            return artists[0]['uri']
+            artist_uri = artists[0]['uri']
 
         elif num_matches > 1:
             if VERBOSE:
                 print ("%i matches found: " % num_matches + str([a['name'] for a in artists]) )
             # check for exact match
             for a in artists:
+                no_match_found =True 
```

Comment:
Yep sorry stupid mistake by me coding like a noob at 1am. I meant to create it before the loop not sure if it exists after but my Python intuition says it would. Either way your change is needed